### PR TITLE
feat(crux-ui): show versions on product edit and on version add

### PR DESCRIPTION
**Pull request by balanceee**

### CODE REVIEW

**Review by polaroi8d — APPROVED** [review]
LGTM

File: web/crux-ui/src/components/products/product-versions-section.tsx [hunk 1]
```diff
@@ -17,18 +18,21 @@ const ProductVersionsSection = (props: ProductVersionsSectionProps) => {
 
   const onClick = (versionId: string) => router.push(versionUrl(props.productId, versionId))
 
-  return (
+  return props.versions ? (
```
Comment (m8vago) [inline web/crux-ui/src/components/products/product-versions-section.tsx:21]: we shouldn't bury deep down null checks.
please set versions to non-nullable, and simply check for null at the place of usage

File: web/crux-ui/src/components/products/versions/version-card.tsx [hunk 2]
```diff
@@ -17,7 +17,8 @@ interface VersionCardProps {
   productId: string
   version: Version
   onClick?: () => void
-  onIncreaseClick: () => void
+  onIncreaseClick?: () => void
+  readonly?: boolean
```
Comment (m8vago) [inline web/crux-ui/src/components/products/versions/version-card.tsx:21]: use `disabled` please

File: web/crux-ui/src/components/products/edit-product-card.tsx [hunk 3]
```diff
@@ -143,6 +145,7 @@ const EditProductCard = (props: EditProductCardProps) => {
           <DyoButton className="hidden" type="submit"></DyoButton>
         </form>
       </DyoCard>
+      <ProductVersionsSection productId={product.id} versions={props.versions} readonly />
```
Comment (robot9706) [inline web/crux-ui/src/components/products/edit-product-card.tsx:148]: In my opinion `ProductVersionsSection` shouldn't handle the undefined case. In `ProductVersionsSection` `versions` should be mandatory and `EditProductCard` should hande the undefined case (for example `props.version ?? []`).

This way `ProductVersionsSection` doesn't need an empty fragment.

**Review by robot9706 — CHANGES_REQUESTED** [review]

**Review by robot9706 — APPROVED** [review]
😎

**Review by m8vago — APPROVED** [review]

**Comment by m8vago** [comment]
Nice work!

**Pull request merged**